chassis/invokers/src/main/java/com/griddynamics/jagger/invoker/http/v2/AbstractHttpInvoker.java

```diff
@@ -0,0 +1,48 @@
+package com.griddynamics.jagger.invoker.http.v2;
+
+import com.griddynamics.jagger.invoker.InvocationException;
+import com.griddynamics.jagger.invoker.Invoker;
+import com.griddynamics.jagger.invoker.http.ApacheAbstractHttpInvoker;
+import com.griddynamics.jagger.invoker.http.HttpInvoker;
+
+/**
+ * An object that represents abstract HTTP-invoker that invokes services of SuT via http protocol. <p>
+ * Extending classes should provide its own implementation of
+ * {@link #invoke(JHttpQuery query, JHttpEndpoint endpoint)} method.<p>
+ * Also, {@link HTTP_CLIENT httpClient} must be provided by constructor {@link AbstractHttpInvoker#AbstractHttpInvoker(HTTP_CLIENT)}.<br/>
+ *
+ * @param <HTTP_CLIENT> the type of the HTTP-client (look at {@link JHttpClient})<p>
+ * @author Anton Antonenko
+ * @see Invoker
+ * @see ApacheAbstractHttpInvoker
+ * @see HttpInvoker
+ * @since 1.3
+ */
+@SuppressWarnings("unused")
+public abstract class AbstractHttpInvoker<HTTP_CLIENT extends JHttpClient> implements Invoker<JHttpQuery, JHttpResponse, JHttpEndpoint> {
```

Comment:
Why such a naming? It is confusing. I thought it is a constant reference, then I thought it is a class name nad finally I understood after a few minutes of investigation